File: exercism/r/grains/grains.R
square <- function(n) {
    if(n < 1 | n > 64){
        stop("Error")
    }
    2^(n - 1)
}

total <- function() {
    vec<-1:64
    sum(2^(vec-1))
    # sum(sapply(1:64, square))
}

# quick test using cli i.e Rscript .\difference-of-squares.R 10
# args <- commandArgs(trailingOnly=TRUE)
# paste0("args=", args)
total()

File: exercism/go/grains/grains.go
package grains

import (
	"errors"
	"math"
)

func Square(number int) (uint64, error) {
	if number < 1 || number > 64 {
		return 0, errors.New("bad")
	}

	return uint64(math.Pow(2.0, float64(number-1))), nil
}

func Total() uint64 {
	var sum uint64
	for i := 1; i < 65; i++ {
		x, _ := Square(i)
		sum += x
	}
	return sum
}
